STX.REST.RESTFulSense.Clients.Tests.Unit/Services/Foundations/HttpExchanges/HttpExchangeServiceTests.Headers.cs

```diff
@@ -0,0 +1,1778 @@
+                from invalidProtocolNameHeader in invalidProtocolNameHeaders
+                from invalidProtocolVersionHeader in invalidProtocolVersionHeaders
+                from invalidReceivedByHeader in invalidReceivedByHeaders
+                select new ViaHeader
+                {
+                    Comment = invalidCommentHeader,
+                    ProtocolName = invalidProtocolNameHeader,
+                    ProtocolVersion = invalidProtocolVersionHeader,
+                    ReceivedBy = invalidReceivedByHeader
+                };
+
+            return invalidViaHeaders
+                .Prepend(null)
+                .ToArray();
+        }
+
+        private static WarningHeader[] CreateInvalidWarningHeaders()
+        {
+            string[] invalidAgentHeaders = CreateInvalidStringArrayHeaders();
+            string[] invalidTextHeaders = CreateInvalidStringArrayHeaders();
+
+            IEnumerable<WarningHeader> invalidViaHeaders =
+                from invalidAgentHeader in invalidAgentHeaders
+                from invalidTextHeader in invalidTextHeaders
+                select new WarningHeader
+                {
+                    Agent = invalidAgentHeader,
+                    Text = invalidTextHeader
+                };
+
+            return invalidViaHeaders
+                .Prepend(null)
+                .ToArray();
+        }
+
+        private static ContentDispositionHeader[] CreateInvalidContentDispositionHeaders()
+        {
+            string[] invalidDispositionTypes = CreateInvalidStringArrayHeaders();
+            string[] invalidNames = CreateInvalidStringArrayHeaders();
+            string[] invalidFileNames = CreateInvalidStringArrayHeaders();
+            string[] invalidFileNameStars = CreateInvalidStringArrayHeaders();
+
+            IEnumerable<ContentDispositionHeader> invalidContentDispositionHeaders =
+                from invalidDispositionType in invalidDispositionTypes
+                from invalidName in invalidNames
+                from invalidFileName in invalidFileNames
+                from invalidFileNameStar in invalidFileNameStars
+                select new ContentDispositionHeader
+                {
+                    DispositionType = invalidDispositionType,
+                    Name = invalidName,
+                    FileName = invalidFileName,
+                    FileNameStar = invalidFileNameStar,
+                };
+
+            return invalidContentDispositionHeaders
+                .Prepend(null)
+                .ToArray();
+        }
+
+        private static ContentRangeHeader[] CreateInvalidContentRangeHeaders()
+        {
+            string[] invalidUnits = CreateInvalidStringArrayHeaders();
+
+            IEnumerable<ContentRangeHeader> invalidContentRangeHeaders =
+                from invalidUnit in invalidUnits
+                select new ContentRangeHeader
+                {
+                    Unit = invalidUnit
```

Comment:
what about the from, to, length?
We know cases that it will fail, for example if from > to. Look the happy test part where we have that logic to initialize valid values.

Reply:
I will fix it